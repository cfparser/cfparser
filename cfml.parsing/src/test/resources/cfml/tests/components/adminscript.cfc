component {
    function foo(){
        admin action="getRegional" type="test" password="test" returnVariable="rtn";
        return returnVariable;
}
}
