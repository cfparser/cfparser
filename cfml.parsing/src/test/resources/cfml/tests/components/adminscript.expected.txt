/*===TOKENS===*/
COMPONENT                <component>
'{'                      <{>
FUNCTION                 <function>
IDENTIFIER               <foo>
'('                      <(>
')'                      <)>
'{'                      <{>
ADMIN                    <admin>
IDENTIFIER               <action>
'='                      <=>
OPEN_STRING              <">
STRING_LITERAL           <getRegional>
CLOSE_STRING             <">
IDENTIFIER               <type>
'='                      <=>
OPEN_STRING              <">
STRING_LITERAL           <test>
CLOSE_STRING             <">
IDENTIFIER               <password>
'='                      <=>
OPEN_STRING              <">
STRING_LITERAL           <test>
CLOSE_STRING             <">
IDENTIFIER               <returnVariable>
'='                      <=>
OPEN_STRING              <">
STRING_LITERAL           <rtn>
CLOSE_STRING             <">
';'                      <;>
RETURN                   <return>
IDENTIFIER               <returnVariable>
';'                      <;>
'}'                      <}>
'}'                      <}>
/*===TREE===*/
(scriptBlock
  (componentDeclaration
    component
    (componentGuts
      {
      (element
        (functionDeclaration
          function
          (identifier foo)
          (
          parameterList
          )
          (compoundStatement
            {
            (statement
              (tagOperatorStatement
                (adminStatement
                  admin
                  (paramStatementAttributes
                    (param
                      (multipartIdentifier (identifier action))
                      =
                      (startExpression
                        (compareExpression
                          (baseExpression
                            (unaryExpression
                              (primaryExpression
                                (literalExpression (stringLiteral " (stringLiteralPart getRegional) "))
                              )
                            )
                          )
                        )
                      )
                    )
                    (param
                      (multipartIdentifier (identifier type))
                      =
                      (startExpression
                        (compareExpression
                          (baseExpression
                            (unaryExpression
                              (primaryExpression
                                (literalExpression (stringLiteral " (stringLiteralPart test) "))
                              )
                            )
                          )
                        )
                      )
                    )
                    (param
                      (multipartIdentifier (identifier password))
                      =
                      (startExpression
                        (compareExpression
                          (baseExpression
                            (unaryExpression
                              (primaryExpression
                                (literalExpression (stringLiteral " (stringLiteralPart test) "))
                              )
                            )
                          )
                        )
                      )
                    )
                    (param
                      (multipartIdentifier (identifier returnVariable))
                      =
                      (startExpression
                        (compareExpression
                          (baseExpression
                            (unaryExpression
                              (primaryExpression
                                (literalExpression (stringLiteral " (stringLiteralPart rtn) "))
                              )
                            )
                          )
                        )
                      )
                    )
                  )
                  ;
                )
              )
            )
            (statement
              (returnStatement
                return
                (startExpression
                  (compareExpression
                    (baseExpression
                      (unaryExpression (memberExpression (identifier returnVariable)))
                    )
                  )
                )
                ;
              )
            )
            }
          )
        )
      )
      }
    )
  )
)
/*======*/
